addons/website/static/src/components/wysiwyg_adapter/wysiwyg_adapter.js

```diff
@@ -646,7 +719,208 @@ export class WysiwygAdapterComponent extends ComponentAdapter {
+            'service_context_get': this._onServiceContextGet.bind(this),
+            'action_demand': this._handleAction.bind(this),
+            'request_cancel': this._onCancelRequest.bind(this),
+            'snippet_will_be_cloned': this._onSnippetWillBeCloned.bind(this),
+            'snippet_cloned': this._onSnippetCloned.bind(this),
+            'snippet_dropped': this._onSnippetDropped.bind(this),
+            'snippet_removed': this._onSnippetRemoved.bind(this),
+            'reload_bundles': this._reloadBundles.bind(this),
+            'menu_dialog': this._onMenuDialogRequest.bind(this),
+            'update_color_previews': this._onColorPreviewsUpdate.bind(this),
+            'request_mobile_preview': this._onMobilePreviewRequest.bind(this),
+            'get_switchable_related_views': this._onGetSwitchableRelatedViews.bind(this),
+        };
+
+
+        const evType = ev.name;
+        const payload = ev.data;
+        if (evType === 'call_service') {
+            let args = payload.args || [];
+            if (payload.service === 'ajax' && payload.method === 'rpc') {
+                // ajax service uses an extra 'target' argument for rpc
+                args = args.concat(ev.target);
+            }
+            const service = legacyEnv.services[payload.service];
+            //If the service doesn't exist it means that it was translated to Owl
+            if (service) {
+                const result = service[payload.method].apply(service, args);
+                payload.callback(result);
+            } else {
+                throw new Error(
+                    `The service "${payload.service}" is not present in the legacy owl environment.
+                        You should probably create a mapper in @web/legacy/utils`
+                );
+            }
+        } else if (evType === 'get_session') {
+            if (payload.callback) {
+                payload.callback(this.env.session);
+            }
+        } else if (evType === 'load_views') {
+            const params = {
+                model: payload.modelName,
+                context: payload.context,
+                views_descr: payload.views,
+            };
+            this.env.dataManager
+                .load_views(params, payload.options || {})
+                .then(payload.on_success);
+        } else if (evType === 'load_filters') {
+            return this.env.dataManager
+                .load_filters(payload)
+                .then(payload.on_success);
+        } else if (evType in triggers) {
+            triggers[evType](ev);
+        }
+    }
+    /**
+     * @override
+     * @returns {Promise}
+     */
+    async _saveViewBlocks() {
+        await super._saveViewBlocks(...arguments);
+        if (this.isDirty()) {
+            return this._restoreMegaMenus();
+        }
+    }
+    /**
+     * @private
+     * @param {HTMLElement} editable
+     */
+    _saveCoverProperties($elementToSave) {
```

Comment:
I get it, all those "new" functions were moved from the `website` `wysiwyg.js` to `website` `wysiwyg_adapter.js`.